helper/schema/errors.go

```diff
@@ -0,0 +1,22 @@
+package schema
+
+import (
+	"github.com/zclconf/go-cty/cty"
+)
+
+type AttributeError struct {
```

Comment:
What you've written here is, I think, essentially the same as `cty.PathError`. Maybe that's okay as a more explicit declaration of intent, but the `cty` version has some helpers like `path.NewErrorf(...)` that makes it easy to construct a new error when you already have a path.

I'd originally imagined having a helper in here to take an old-style attribute path + `error` and turn it into a `cty.PathError` by first parsing the old-style path, but since you're taking the direction of teaching the SDK about `cty.Path` instead (which is nice as it's a more direct representation) I'm not sure even that is really needed now. :thinking:

Reply:
You're right. I originally had `AttributeError` with `string` based path before I decided to refactor and I didn't realize it's not necessary anymore. 😃 

I'm not sure if we want to go down that road (due to complexity and potentially size of the snippet), but I'm thinking we might introduce a new error type or modify `cty.PathError` in the future to accept multiple paths, so we can render accurate and nice snippets even for errors from `ConflictsWith` where error relates to 2+ fields.